File: packages/shared/src/types/growthAudit.ts
export type GrowthAuditProvider = "openai" | "claude";
export type GrowthAuditMode = "full_site_mvp";
export type GrowthAuditRunStatus =
  | "pending"
  | "queued"
  | "discovering"
  | "classifying"
  | "analyzing"
  | "ready_for_analysis"
  | "completed"
  | "failed"
  | "partial_failed"
  | "cancelled";
export type GrowthAuditPhase =
  | "queued"
  | "discovery"
  | "classification"
  | "analysis"
  | "technical_scan"
  | "page_rescan"
  | "ready_for_analysis"
  | "finalization"
  | "completed"
  | "failed";
export type GrowthAuditPageType =
  | "homepage"
  | "product"
  | "collection"
  | "blog"
  | "blog_article"
  | "article"
  | "static_page"
  | "page"
  | "policy"
  | "cart"
  | "checkout"
  | "search"
  | "account"
  | "other"
  | "unknown";
export type GrowthAuditPageStatus =
  | "pending"
  | "discovered"
  | "classified"
  | "analyzing"
  | "analyzed"
  | "failed"
  | "skipped";
export type GrowthAuditPageSource =
  | "seed"
  | "sitemap"
  | "shopify_product"
  | "shopify_collection"
  | "shopify_page"
  | "shopify_blog"
  | "crawl"
  | "manual";

export type GrowthAuditInventoryFilter =
  | "all"
  | "homepage"
  | "product"
  | "collection"
  | "blog"
  | "static_page"
  | "unknown";

export type GrowthAuditScoreFilter = "all" | "critical" | "warning" | "good";

export type GrowthAuditPageStatusFilter = "all" | "analyzed" | "failed";

export type GrowthAuditFindingSeverity =
  | "critical"
  | "high"
  | "medium"
  | "low"
  | "info";

export type GrowthAuditFindingCategory =
  | "technical"
  | "seo"
  | "schema"
  | "images"
  | "content"
  | "geo"
  | "cro"
  | "ads";

export type GrowthAuditTaskOwnerType = "seo" | "content" | "dev" | "design" | "ads";

export type GrowthAuditFindingStatus = "open" | "completed" | "dismissed" | "superseded";

export type GrowthAuditTaskStatus = "open" | "completed" | "dismissed" | "superseded";

export interface GrowthAuditRunSummary {
  message?: string;
  pagesDiscovered?: number;
  pagesClassified?: number;
  pagesAnalyzed?: number;
  pagesFailed?: number;
  averageTechnicalScore?: number | null;
  criticalFindings?: number;
  highFindings?: number;
  tasksOpen?: number;
  includeAiAnalysis?: boolean;
  auditMode?: string;
  sources?: {
    seed?: number;
    sitemap?: number;
    shopify?: number;
  };
  pageTypes?: Record<string, number>;
  nextStep?: string;
  warning?: string | null;
  lastPageRescanAt?: string | null;
  aiPagesAnalyzed?: number;
  geoFindings?: number;
  croFindings?: number;
  adsFindings?: number;
  lastAiAnalysisAt?: string | null;
  lastAiAnalysisUrl?: string | null;
  performancePagesAnalyzed?: number;
  averagePerformanceScore?: number | null;
  performanceIssuesOpen?: number;
  lastPerformanceAnalysisAt?: string | null;
  lastPerformanceAnalysisUrl?: string | null;
  searchConsole?: GrowthAuditRunSearchConsoleSummary;
  analytics?: GrowthAuditRunAnalyticsSummary;
  shopifyCommerce?: GrowthAuditRunShopifyCommerceSummary;
  ga4Ecommerce?: GrowthAuditRunGa4EcommerceSummary;
}

export interface GrowthAuditRunAnalyticsSummary {
  totalSessions?: number;
  totalUsers?: number;
  averageEngagementRate?: number;
  totalConversions?: number;
  totalRevenue?: number;
  pagesWithData?: number;
  lowEngagementPages?: number;
  highTrafficLowConversionPages?: number;
  lastSyncedAt?: string | null;
}

export interface GrowthAuditRunSearchConsoleSummary {
  totalClicks?: number;
  totalImpressions?: number;
  averageCtr?: number;
  averagePosition?: number;
  pagesWithData?: number;
  opportunityPages?: number;
  lastSyncedAt?: string | null;
}

export interface GrowthAuditPageSearchConsoleQuery {
  query: string;
  clicks?: number;
  impressions?: number;
  ctr?: number;
  position?: number;
}

export interface GrowthAuditPageSearchConsoleMetadata {
  clicks?: number;
  impressions?: number;
  ctr?: number;
  position?: number;
  topQueries?: GrowthAuditPageSearchConsoleQuery[];
  syncedAt?: string;
}

export interface GrowthAuditPageAnalyticsMetadata {
  sessions?: number;
  totalUsers?: number;
  engagedSessions?: number;
  engagementRate?: number;
  averageSessionDuration?: number;
  conversions?: number;
  revenue?: number;
  source?: string;
  periodDays?: number;
  syncedAt?: string;
}

export interface GrowthAuditPageShopifyCommerceMetadata {
  periodDays?: number;
  quantitySold?: number;
  ordersCount?: number;
  sales?: number;
  currency?: string;
  averageUnitPrice?: number;
  averageOrderValue?: number;
  stock?: number | null;
  availableForSale?: boolean | null;
  productStatus?: string | null;
  priceMin?: number | null;
  priceMax?: number | null;
  syncedAt?: string;
}

export interface GrowthAuditRunShopifyCommerceSummary {
  periodDays?: number;
  totalSales?: number;
  totalQuantitySold?: number;
  productsWithSales?: number;
  productsWithoutSales?: number;
  productsOutOfStock?: number;
  currency?: string;
  topProducts?: Array<{
    pageId: string;
    productGid?: string | null;
    title?: string | null;
    sales?: number;
    quantitySold?: number;
    ordersCount?: number;
  }>;
  lastSyncedAt?: string | null;
}

export interface GrowthAuditGa4MatchCandidateItem {
  itemId?: string;
  itemName?: string;
  itemVariant?: string;
  itemsViewed?: number;
  itemsAddedToCart?: number;
  itemsPurchased?: number;
  itemRevenue?: number;
  candidateReason?: string;
}

export interface GrowthAuditGa4MatchShopifyKeys {
  productGid?: string;
  productLegacyId?: string | null;
  variantLegacyIds?: string[];
  skus?: string[];
  titleNormalized?: string;
  handleNormalized?: string;
}

export interface GrowthAuditGa4MatchDebug {
  shopifyKeys: GrowthAuditGa4MatchShopifyKeys;
  matchedBy?: string;
  matchStatus: "matched" | "no_reliable_match" | "ambiguous_match";
  reason: string;
  candidateItems: GrowthAuditGa4MatchCandidateItem[];
}

export interface GrowthAuditPageGa4EcommerceMetadata {
  periodDays?: number;
  itemViews?: number;
  itemViewEvents?: number;
  itemsAddedToCart?: number;
  itemsCheckedOut?: number;
  itemsPurchased?: number;
  itemRevenue?: number;
  currency?: string;
  viewToCartRate?: number;
  cartToCheckoutRate?: number;
  checkoutToPurchaseRate?: number;
  viewToPurchaseRate?: number;
  cartToPurchaseRate?: number;
  dropoffViewToCart?: number;
  dropoffCartToCheckout?: number;
  dropoffCheckoutToPurchase?: number;
  matchedBy?: string;
  matchedItemIds?: string[];
  matchedItemNames?: string[];
  matchDebug?: GrowthAuditGa4MatchDebug;
  source?: string;
  syncedAt?: string;
}

export interface GrowthAuditRunGa4EcommerceSummary {
  periodDays?: number;
  totalItemViews?: number;
  totalItemsAddedToCart?: number;
  totalItemsCheckedOut?: number;
  totalItemsPurchased?: number;
  totalItemRevenue?: number;
  averageViewToCartRate?: number;
  averageCartToPurchaseRate?: number;
  productsWithFunnelData?: number;
  productsWithoutFunnelData?: number;
  unmatchedItems?: number;
  matchedProducts?: number;
  productsWithNoReliableMatch?: number;
  ambiguousItemsCount?: number;
  matchingMode?: string;
  matchingWarning?: string;
  highViewLowCartProducts?: number;
  highCartLowPurchaseProducts?: number;
  topFunnelProducts?: Array<{
    pageId: string;
    title?: string | null;
    itemViews?: number;
    itemsAddedToCart?: number;
    itemsPurchased?: number;
    itemRevenue?: number;
  }>;
  lastSyncedAt?: string | null;
  currency?: string;
}

export interface GrowthAuditPageAiMetadata {
  latestResultId?: string;
  latestScore?: number;
  seoScore?: number;
  geoScore?: number;
  croScore?: number;
  adsReadinessScore?: number;
  analyzedAt?: string;
}

export interface GrowthAuditPagePerformanceMetadata {
  latestResultId?: string;
  latestScore?: number;
  analyzedAt?: string;
  cruxSource?: string | null;
  lcp?: number | null;
  cls?: number | null;
  inp?: number | null;
  strategy?: string;
}

export interface GrowthAuditRunCreateRequest {
  rootUrl: string;
  provider?: GrowthAuditProvider;
  auditMode?: GrowthAuditMode;
  maxPages?: number;
  includeAiAnalysis?: boolean;
}

export interface GrowthAuditRun {
  id: string;
  projectId: string;
  rootUrl: string;
  normalizedDomain: string;
  status: GrowthAuditRunStatus;
  phase?: string | null;
  auditMode: string;
  provider: GrowthAuditProvider;
  progressPercent: number;
  pagesDiscovered: number;
  pagesClassified: number;
  pagesAnalyzed: number;
  pagesFailed: number;
  totalPages?: number | null;
  currentUrl?: string | null;
  config?: Record<string, unknown> | null;
  summary?: GrowthAuditRunSummary | null;
  siteScore?: number | null;
  seoScore?: number | null;
  geoScore?: number | null;
  croScore?: number | null;
  performanceScore?: number | null;
  errorMessage?: string | null;
  startedAt?: string | null;
  completedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export type GrowthAuditSourceEntityType =
  | "shopify_product"
  | "shopify_collection"
  | "shopify_page"
  | "shopify_article";

export interface GrowthAuditPage {
  id: string;
  runId: string;
  projectId: string;
  url: string;
  normalizedUrl: string;
  path?: string | null;
  pageType: GrowthAuditPageType | string;
  source: GrowthAuditPageSource | string;
  status: GrowthAuditPageStatus | string;
  priority: string;
  title?: string | null;
  metaDescription?: string | null;
  canonicalUrl?: string | null;
  h1?: string | null;
  httpStatus?: number | null;
  depth?: number | null;
  score?: number | null;
  seoScore?: number | null;
  geoScore?: number | null;
  croScore?: number | null;
  performanceScore?: number | null;
  discoveredAt?: string | null;
  classifiedAt?: string | null;
  analyzedAt?: string | null;
  errorMessage?: string | null;
  metadata?: Record<string, unknown> | null;
  sourceEntityType?: GrowthAuditSourceEntityType | string | null;
  sourceEntityId?: string | null;
  sourceEntityGid?: string | null;
  sourceEntityHandle?: string | null;
  sourceEntityTitle?: string | null;
  sourceEntitySyncedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface GrowthAuditEvent {
  id: string;
  runId: string;
  projectId: string;
  eventType: string;
  phase?: string | null;
  message: string;
  progressPercent?: number | null;
  payload?: Record<string, unknown> | null;
  createdAt?: string | null;
}

export interface GrowthAuditInventoryCounts {
  total: number;
  homepage: number;
  product: number;
  collection: number;
  blog: number;
  staticPage: number;
  unknown: number;
  bySource: Record<string, number>;
  byStatus: Record<string, number>;
}

export interface GrowthAuditRunDetailResponse {
  run: GrowthAuditRun;
  pages: GrowthAuditPage[];
  events: GrowthAuditEvent[];
  findingsCount: number;
  tasksCount: number;
}

export interface GrowthAuditRunsListResponse {
  runs: GrowthAuditRun[];
}

export interface GrowthAuditPagesListResponse {
  pages: GrowthAuditPage[];
}

export interface GrowthAuditEventsListResponse {
  events: GrowthAuditEvent[];
}

export interface GrowthAuditStartResponse {
  run: GrowthAuditRun;
}

export interface GrowthAuditFinding {
  id: string;
  runId: string;
  pageId?: string | null;
  projectId: string;
  sourceResultId?: string | null;
  category: GrowthAuditFindingCategory | string;
  severity: GrowthAuditFindingSeverity | string;
  priority: string;
  title: string;
  description?: string | null;
  evidence?: string | null;
  recommendation?: string | null;
  howToValidate?: string | null;
  impact?: string | null;
  effort?: string | null;
  status: string;
  metadata?: Record<string, unknown> | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface GrowthAuditTask {
  id: string;
  runId: string;
  pageId?: string | null;
  findingId?: string | null;
  projectId: string;
  title: string;
  description?: string | null;
  ownerType: GrowthAuditTaskOwnerType | string;
  priority: string;
  estimatedEffort: string;
  status: string;
  metadata?: Record<string, unknown> | null;
  completedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface GrowthAuditFindingsListResponse {
  findings: GrowthAuditFinding[];
}

export interface GrowthAuditTasksListResponse {
  tasks: GrowthAuditTask[];
}

export interface GrowthAuditFindingsFilters {
  pageId?: string;
  severity?: string;
  category?: string;
  status?: string;
}

export interface GrowthAuditTasksFilters {
  pageId?: string;
  priority?: string;
  ownerType?: string;
  status?: string;
}

export interface GrowthAuditPageRescanRequest {
  clearPreviousOpenItems?: boolean;
  note?: string | null;
}

export interface GrowthAuditPageRescanResponse {
  run: GrowthAuditRun;
  page: GrowthAuditPage;
  findingsCount: number;
  tasksCount: number;
  message: string;
}

export type GrowthAuditAiAnalysisDepth = "standard" | "deep";

export interface GrowthAuditPageResult {
  id: string;
  runId: string;
  pageId: string;
  projectId: string;
  resultType: string;
  skillKey?: string | null;
  status: string;
  score?: number | null;
  summary?: string | null;
  findings?: Array<Record<string, unknown>> | null;
  recommendations?: Array<Record<string, unknown>> | null;
  tasks?: Array<Record<string, unknown>> | null;
  artifacts?: Record<string, unknown> | null;
  rawOutput?: Record<string, unknown> | null;
  errorMessage?: string | null;
  startedAt?: string | null;
  completedAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
}

export interface GrowthAuditPageAiAnalysisRequest {
  provider?: GrowthAuditProvider;
  depth?: GrowthAuditAiAnalysisDepth;
  includeSeo?: boolean;
  includeGeo?: boolean;
  includeCro?: boolean;
  includeAdsReadiness?: boolean;
  note?: string | null;
}

export interface GrowthAuditPageAiAnalysisResponse {
  run: GrowthAuditRun;
  page: GrowthAuditPage;
  result: GrowthAuditPageResult;
  findingsCount: number;
  tasksCount: number;
  message: string;
}

export interface GrowthAuditPagePerformanceAnalysisRequest {
  strategy?: "mobile" | "desktop";
}

export interface GrowthAuditPagePerformanceAnalysisResponse {
  run: GrowthAuditRun;
  page: GrowthAuditPage;
  result: GrowthAuditPageResult;
  findingsCount: number;
  tasksCount: number;
  message: string;
}

export interface GrowthAuditSearchConsoleAnalysisRequest {
  days?: number;
}

export interface GrowthAuditSearchConsoleAnalysisResponse {
  run: GrowthAuditRun;
  summary: GrowthAuditRunSearchConsoleSummary;
  message: string;
}

export interface GrowthAuditAnalyticsAnalysisRequest {
  days?: number;
}

export interface GrowthAuditAnalyticsAnalysisResponse {
  run: GrowthAuditRun;
  summary: GrowthAuditRunAnalyticsSummary;
  message: string;
}

export interface GrowthAuditShopifyCommerceAnalysisRequest {
  days?: number;
}

export interface GrowthAuditShopifyCommerceAnalysisResponse {
  run: GrowthAuditRun;
  summary: GrowthAuditRunShopifyCommerceSummary;
  message: string;
}

export interface GrowthAuditGa4EcommerceAnalysisRequest {
  days?: number;
}

export interface GrowthAuditGa4EcommerceAnalysisResponse {
  run: GrowthAuditRun;
  summary: GrowthAuditRunGa4EcommerceSummary;
  message: string;
}

export interface GrowthAuditPageResultsListResponse {
  results: GrowthAuditPageResult[];
}

File: apps/api/app/schemas/growth_audit.py
from datetime import datetime
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


GrowthAuditProvider = Literal["openai", "claude"]
GrowthAuditMode = Literal["full_site_mvp"]
GrowthAuditRunStatus = Literal[
    "pending",
    "queued",
    "discovering",
    "classifying",
    "analyzing",
    "ready_for_analysis",
    "completed",
    "failed",
    "partial_failed",
    "cancelled",
]
GrowthAuditPhase = Literal[
    "queued",
    "discovery",
    "classification",
    "analysis",
    "technical_scan",
    "ready_for_analysis",
    "finalization",
    "completed",
    "failed",
]
GrowthAuditPageType = Literal[
    "homepage",
    "product",
    "collection",
    "blog",
    "blog_article",
    "article",
    "static_page",
    "page",
    "policy",
    "cart",
    "checkout",
    "search",
    "account",
    "other",
    "unknown",
]
GrowthAuditPageStatus = Literal[
    "pending",
    "discovered",
    "classified",
    "analyzing",
    "analyzed",
    "failed",
    "skipped",
]
GrowthAuditPageSource = Literal[
    "seed",
    "sitemap",
    "shopify_product",
    "shopify_collection",
    "shopify_page",
    "shopify_blog",
    "crawl",
    "manual",
]


class GrowthAuditRunCreateRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    root_url: str = Field(alias="rootUrl")
    provider: GrowthAuditProvider = "openai"
    audit_mode: GrowthAuditMode = Field(default="full_site_mvp", alias="auditMode")
    max_pages: int = Field(default=50, alias="maxPages")
    include_ai_analysis: bool = Field(default=False, alias="includeAiAnalysis")


class GrowthAuditRunRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    project_id: UUID = Field(serialization_alias="projectId")
    root_url: str = Field(serialization_alias="rootUrl")
    normalized_domain: str = Field(serialization_alias="normalizedDomain")
    status: str
    phase: str | None = None
    audit_mode: str = Field(serialization_alias="auditMode")
    provider: GrowthAuditProvider
    progress_percent: int = Field(serialization_alias="progressPercent")
    pages_discovered: int = Field(serialization_alias="pagesDiscovered")
    pages_classified: int = Field(serialization_alias="pagesClassified")
    pages_analyzed: int = Field(serialization_alias="pagesAnalyzed")
    pages_failed: int = Field(serialization_alias="pagesFailed")
    total_pages: int | None = Field(default=None, serialization_alias="totalPages")
    current_url: str | None = Field(default=None, serialization_alias="currentUrl")
    config: dict[str, Any] | None = None
    summary: dict[str, Any] | None = None
    site_score: int | None = Field(default=None, serialization_alias="siteScore")
    seo_score: int | None = Field(default=None, serialization_alias="seoScore")
    geo_score: int | None = Field(default=None, serialization_alias="geoScore")
    cro_score: int | None = Field(default=None, serialization_alias="croScore")
    performance_score: int | None = Field(
        default=None,
        serialization_alias="performanceScore",
    )
    error_message: str | None = Field(default=None, serialization_alias="errorMessage")
    started_at: datetime | None = Field(default=None, serialization_alias="startedAt")
    completed_at: datetime | None = Field(default=None, serialization_alias="completedAt")
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")


class GrowthAuditPageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    run_id: UUID = Field(serialization_alias="runId")
    project_id: UUID = Field(serialization_alias="projectId")
    url: str
    normalized_url: str = Field(serialization_alias="normalizedUrl")
    path: str | None = None
    page_type: str = Field(serialization_alias="pageType")
    source: str
    status: str
    priority: str
    title: str | None = None
    meta_description: str | None = Field(
        default=None,
        serialization_alias="metaDescription",
    )
    canonical_url: str | None = Field(default=None, serialization_alias="canonicalUrl")
    h1: str | None = None
    http_status: int | None = Field(default=None, serialization_alias="httpStatus")
    depth: int | None = None
    score: int | None = None
    seo_score: int | None = Field(default=None, serialization_alias="seoScore")
    geo_score: int | None = Field(default=None, serialization_alias="geoScore")
    cro_score: int | None = Field(default=None, serialization_alias="croScore")
    performance_score: int | None = Field(
        default=None,
        serialization_alias="performanceScore",
    )
    discovered_at: datetime | None = Field(default=None, serialization_alias="discoveredAt")
    classified_at: datetime | None = Field(default=None, serialization_alias="classifiedAt")
    analyzed_at: datetime | None = Field(default=None, serialization_alias="analyzedAt")
    error_message: str | None = Field(default=None, serialization_alias="errorMessage")
    page_metadata: dict[str, Any] | None = Field(
        default=None,
        serialization_alias="metadata",
    )
    source_entity_type: str | None = Field(
        default=None,
        serialization_alias="sourceEntityType",
    )
    source_entity_id: UUID | None = Field(
        default=None,
        serialization_alias="sourceEntityId",
    )
    source_entity_gid: str | None = Field(
        default=None,
        serialization_alias="sourceEntityGid",
    )
    source_entity_handle: str | None = Field(
        default=None,
        serialization_alias="sourceEntityHandle",
    )
    source_entity_title: str | None = Field(
        default=None,
        serialization_alias="sourceEntityTitle",
    )
    source_entity_synced_at: datetime | None = Field(
        default=None,
        serialization_alias="sourceEntitySyncedAt",
    )
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")


class GrowthAuditEventRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    run_id: UUID = Field(serialization_alias="runId")
    project_id: UUID = Field(serialization_alias="projectId")
    event_type: str = Field(serialization_alias="eventType")
    phase: str | None = None
    message: str
    progress_percent: int | None = Field(
        default=None,
        serialization_alias="progressPercent",
    )
    payload: dict[str, Any] | None = None
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")


class GrowthAuditRunDetailResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    pages: list[GrowthAuditPageRead]
    events: list[GrowthAuditEventRead]
    findings_count: int = Field(serialization_alias="findingsCount")
    tasks_count: int = Field(serialization_alias="tasksCount")


class GrowthAuditRunsListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    runs: list[GrowthAuditRunRead]


class GrowthAuditPagesListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    pages: list[GrowthAuditPageRead]


class GrowthAuditEventsListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    events: list[GrowthAuditEventRead]


class GrowthAuditStartResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead


class GrowthAuditFindingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    run_id: UUID = Field(serialization_alias="runId")
    page_id: UUID | None = Field(default=None, serialization_alias="pageId")
    project_id: UUID = Field(serialization_alias="projectId")
    source_result_id: UUID | None = Field(
        default=None,
        serialization_alias="sourceResultId",
    )
    category: str
    severity: str
    priority: str
    title: str
    description: str | None = None
    evidence: str | None = None
    recommendation: str | None = None
    how_to_validate: str | None = Field(
        default=None,
        serialization_alias="howToValidate",
    )
    impact: str | None = None
    effort: str | None = None
    status: str
    finding_metadata: dict[str, Any] | None = Field(
        default=None,
        serialization_alias="metadata",
    )
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")


class GrowthAuditTaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    run_id: UUID = Field(serialization_alias="runId")
    page_id: UUID | None = Field(default=None, serialization_alias="pageId")
    finding_id: UUID | None = Field(default=None, serialization_alias="findingId")
    project_id: UUID = Field(serialization_alias="projectId")
    title: str
    description: str | None = None
    owner_type: str = Field(serialization_alias="ownerType")
    priority: str
    estimated_effort: str = Field(serialization_alias="estimatedEffort")
    status: str
    task_metadata: dict[str, Any] | None = Field(
        default=None,
        serialization_alias="metadata",
    )
    completed_at: datetime | None = Field(default=None, serialization_alias="completedAt")
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")


class GrowthAuditFindingsListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    findings: list[GrowthAuditFindingRead]


class GrowthAuditTasksListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    tasks: list[GrowthAuditTaskRead]


class GrowthAuditPageRescanRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    clear_previous_open_items: bool = Field(
        default=True,
        alias="clearPreviousOpenItems",
    )
    note: str | None = None


class GrowthAuditPageRescanResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    page: GrowthAuditPageRead
    findings_count: int = Field(serialization_alias="findingsCount")
    tasks_count: int = Field(serialization_alias="tasksCount")
    message: str


GrowthAuditAiAnalysisDepth = Literal["standard", "deep"]


class GrowthAuditPageResultRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: UUID
    run_id: UUID = Field(serialization_alias="runId")
    page_id: UUID = Field(serialization_alias="pageId")
    project_id: UUID = Field(serialization_alias="projectId")
    result_type: str = Field(serialization_alias="resultType")
    skill_key: str | None = Field(default=None, serialization_alias="skillKey")
    status: str
    score: int | None = None
    summary: str | None = None
    findings: list[Any] | None = None
    recommendations: list[Any] | None = None
    tasks: list[Any] | None = None
    artifacts: dict[str, Any] | None = None
    raw_output: dict[str, Any] | None = Field(default=None, serialization_alias="rawOutput")
    error_message: str | None = Field(default=None, serialization_alias="errorMessage")
    started_at: datetime | None = Field(default=None, serialization_alias="startedAt")
    completed_at: datetime | None = Field(default=None, serialization_alias="completedAt")
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")
    updated_at: datetime | None = Field(default=None, serialization_alias="updatedAt")


class GrowthAuditPageAiAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    provider: GrowthAuditProvider = "openai"
    depth: GrowthAuditAiAnalysisDepth = "standard"
    include_seo: bool = Field(default=True, alias="includeSeo")
    include_geo: bool = Field(default=True, alias="includeGeo")
    include_cro: bool = Field(default=True, alias="includeCro")
    include_ads_readiness: bool = Field(default=True, alias="includeAdsReadiness")
    note: str | None = None


class GrowthAuditPageAiAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    page: GrowthAuditPageRead
    result: GrowthAuditPageResultRead
    findings_count: int = Field(serialization_alias="findingsCount")
    tasks_count: int = Field(serialization_alias="tasksCount")
    message: str


class GrowthAuditPagePerformanceAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    strategy: Literal["mobile", "desktop"] = "mobile"


class GrowthAuditPagePerformanceAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    page: GrowthAuditPageRead
    result: GrowthAuditPageResultRead
    findings_count: int = Field(serialization_alias="findingsCount")
    tasks_count: int = Field(serialization_alias="tasksCount")
    message: str


class GrowthAuditSearchConsoleAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    days: int = Field(default=28, ge=1, le=90)


class GrowthAuditSearchConsoleAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    summary: dict[str, Any]
    message: str


class GrowthAuditAnalyticsAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    days: int = Field(default=28, ge=1, le=90)


class GrowthAuditAnalyticsAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    summary: dict[str, Any]
    message: str


class GrowthAuditShopifyCommerceAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    days: int = Field(default=30, ge=7, le=90)


class GrowthAuditShopifyCommerceAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    summary: dict[str, Any]
    message: str


class GrowthAuditGa4EcommerceAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    days: int = Field(default=30, ge=7, le=90)


class GrowthAuditGa4EcommerceAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    summary: dict[str, Any]
    message: str


class GrowthAuditMerchantCenterAnalysisRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)


class GrowthAuditMerchantCenterAnalysisResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    run: GrowthAuditRunRead
    summary: dict[str, Any]
    message: str


class GrowthAuditPageResultsListResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    results: list[GrowthAuditPageResultRead]
